homeassistant/components/vizio/media_player.py

```diff
@@ -294,6 +338,14 @@ async def async_update_setting(
             setting_type, setting_name, new_value,
         )
 
+    async def async_fetch_latest_apps(self, time: datetime = None) -> None:
+        """Update apps list when fetch_latest_apps service is called."""
+        # Fallback to static list of apps when apps can't be retrieved from external source
+        self._all_known_apps = (
+            await gen_apps_list_from_url(session=async_get_clientsession(self.hass))
```

Comment:
Is this fetched for each entity or shared across them? Maybe consider using a DataUpdateCoordinator to coordinate this ?

Reply:
This is fetched for each entity. DataUpdateCoordinator is at a config entry level right? Is there a way to introduce a DataUpdateCoordinator at the integration level?

Reply:
Is it as simple as creating a new coordinator in `async_setup` and just making sure it gets destroyed on `unload`?

Reply:
Went ahead and went with that approach ^